corgidrp/calibrate_darks_lsq.py

```diff
@@ -0,0 +1,504 @@
+import numpy as np
+import warnings
+from astropy.io import fits
+
+from corgidrp.detector import slice_section, imaging_slice, imaging_area_geom, detector_areas
+import corgidrp.check as check
+from corgidrp.mean_combine import mean_combine
+from corgidrp.data import NoiseMap, BiasOffset
+
+
+class CalDarksLSQException(Exception):
+    """Exception class for calibrate_darks_lsq."""
+
+def calibrate_darks_lsq(datasets, detector_params, d_areas=detector_areas):
```

Comment:
Throughout this PR it would be great to have variable names be more descriptive. It should be a quick find and replace. Some that caught my attention are d_area, all the F D C array-like things, that g_arr, t_arr, k_arr. The motivation here is future maintenance and code readability.